idaes/models/properties/modular_properties/transport_properties/no_method.py

```diff
@@ -0,0 +1,63 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+"Argh!" shouted the cyclops, "IDAES is throwing an error!"
+"Give us the stack trace," replied the developers, "Which method is throwing an error?"
+"NoMethod is!" roared the cyclops, "NoMethod is throwing the error!"
+"Don't waste our time," said the developers, "if no method is throwing an error, you
+don't need our help."
+
+And so Codysseus and his crew were able to sneak out of the cyclops cave and embark
+once more on the wine-dark sea.
+"""
+
+from pyomo.environ import Expression
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+class NoMethod(object):
+    """Class to pass when user wants to deliberately not assign a property to a given phase"""
+
+    class visc_d_phase(object):
```

Comment:
Do we need the nested classes here? I think we should be able to just use a single common class for all properties which would make this far more extensible (so we don't need to modify this every time we add a new property).

Reply:
Looking at `get_phase_method`, it's possible we'd be able to get away with one large class except for one hangup: we need to log which property we're skipping construction of.  Right now, that's being done in the `build_parameters` method of each individual `visc_d_phase`, etc, changing the logged warning as appropriate. It's not obvious to me how we could log an informative message while also using a single common class.

Reply:
I see the issue now - the `build_parameters` method does not get the property as an argument so it can't add the name. Whilst the `get_expression` method does have the necessary information, you don't want to spam this message every time we do a skip (which could be multiple times for the same `IndexedExpression`).

Reply:
I don't think it would spam the log each time each time the property was accessed, just the first time the property was attempted to be constructed (in any phase, not just the one being skipped). However, for something like the MEA column, that would result in a log message for each finite element, so hundreds of log messages. We could get around this by demoting the logged message from "Warning" to "Debug", though.

Reply:
It would still put out a log message for each index as it is constructed, which could be a lot. But maybe that is OK if it was a debug message.

We could consider both - a high level, generic "There is a property with a skipped expression" message at info or warning level (emitted once by `build_parameters`), and a more specific "This index was skipped for property Y" at the debug level (emitted by `get_expression`). That could be done with a single, common class.